epochStart/bootstrap/syncEpochStartMeta.go

```diff
@@ -0,0 +1,125 @@
+package bootstrap
+
+import (
+	"math"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/epochStart"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+)
+
+type epochStartMetaSyncer struct {
+	requestHandler                 epochStart.RequestHandler
+	messenger                      p2p.Messenger
+	epochStartMetaBlockInterceptor EpochStartInterceptor
+	marshalizer                    marshal.Marshalizer
+	hasher                         hashing.Hasher
+}
+
+// ArgsNewEpochStartMetaSyncer -
+type ArgsNewEpochStartMetaSyncer struct {
+	RequestHandler epochStart.RequestHandler
+	Messenger      p2p.Messenger
+	Marshalizer    marshal.Marshalizer
+	Hasher         hashing.Hasher
+}
+
+const delayBetweenRequests = 1 * time.Second
+const thresholdForConsideringMetaBlockCorrect = 0.2
+const numRequestsToSendOnce = 4
+const maxNumTimesToRetry = 100
+
+func NewEpochStartMetaSyncer(args ArgsNewEpochStartMetaSyncer) (*epochStartMetaSyncer, error) {
+	e := &epochStartMetaSyncer{
+		requestHandler: args.RequestHandler,
+		messenger:      args.Messenger,
+		marshalizer:    args.Marshalizer,
+		hasher:         args.Hasher,
+	}
+
+	var err error
+	e.epochStartMetaBlockInterceptor, err = NewSimpleEpochStartMetaBlockInterceptor(e.marshalizer, e.hasher)
+	if err != nil {
+		return nil, err
+	}
+
+	return e, nil
+}
+
+// SyncEpochStartMeta syncs the latest epoch start metablock
+func (e *epochStartMetaSyncer) SyncEpochStartMeta(waitTime time.Duration) (*block.MetaBlock, error) {
+	err := e.initTopicForEpochStartMetaBlockInterceptor()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		e.resetTopicsAndInterceptors()
+	}()
+
+	e.requestEpochStartMetaBlock()
+
+	unknownEpoch := uint32(math.MaxUint32)
+	count := 0
+	for {
+		if count > maxNumTimesToRetry {
+			panic("can't sync with other peers")
+		}
+		count++
+		numConnectedPeers := len(e.messenger.Peers())
```

Comment:
ConnectedPeers instead of Peers (peers are known peers)

Reply:
done.